[HOPS-769] Block report hash mismatch when files are deleted.

smkniazi opened the pull request:
## Make sure there is no duplicate PR for this issue

* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added and passed (for bug fixes / features)
- [x] HOPS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit
- [x] The commit message has the following format: [HOPS-XXX] message

* **Post a link to the associated JIRA issue**
https://hopshadoop.atlassian.net/browse/HOPS-769

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the new behavior (if this is a feature change)?**


* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

HopsworksJenkins commented:
Can one of the admins verify this patch?

berthoug commented on hadoop-hdfs-project/hadoop-hdfs/build/test/data/temp/decommission/exclude:
```diff
@@ -1 +1 @@
-127.0.0.1:36079
```
can you remove this file from the commits, it is a test file and should probably not be committed at all.

berthoug commented on hadoop-hdfs-project/hadoop-hdfs/src/main/java/io/hops/transaction/lock/LockFactory.java:
```diff
@@ -297,7 +296,11 @@ public Lock getHashBucketLock(int storageId) {
   }
-  
+
+  public Lock lockAllUsedHashBuckets() {
```
can you use the same naming convention as for the remaining of the class: getHasBucketsLocksAllFileBlocks, it make it easier to use later on.

berthoug commented:
Jenkins test this please.

berthoug commented:
Envthough there is a merge conflict it run the test on your branch and it looks like there is some tests failing, can you look into it: https://bbc1.sics.se/jenkins/view/pull/job/hops-pull/144/

berthoug commented:
Jenkins test this please.

smkniazi commented:
Tests failed because file overwrite is not properly  handled. I have fixed the code and added couple of new tests.

berthoug commented:
Jenkins test this please.

berthoug commented:
The following tests are still failing: 
org.apache.hadoop.hdfs.TestPersistBlocks
org.apache.hadoop.hdfs.TestAbandonBlock

hadoop-hdfs-project/hadoop-hdfs/build/test/data/temp/decommission/exclude is still there and hadoop-hdfs-project/hadoop-hdfs/build/test/data/temp/decommission/.exclude.crc as been added to the commit are they really needed.

smkniazi commented:
"../exclude" "../exclude.crc"  were there in delete mode. Some previous commit has added them to the project. I have removed them from the commit now

I have fixed the unit tests

berthoug commented:
Jenkins test this please.
